lib/writeGltf.js

```diff
@@ -38,40 +28,93 @@ function writeGltf(gltf, outputPath, isEmbedded, createDirectory, callback) {
                     throw err;
                 }
             }
+            else{
+                fs.writeFile(outputPath, JSON.stringify(gltf, undefined, 2), function (err) {
+                    if (err) {
+                        throw err;
+                    }
+                    if (callback) {
+                        callback();
+                    }
+                });     
+            }   
+        }
+    );
+}
 
-            fs.writeFile(outputPath, JSON.stringify(gltf, undefined, 2), function (err) {
-                if (err) {
-                    throw err;
+function writeSource(gltf, filePath, name, isEmbedded, callback) {
+    var objects = gltf[name];
+
+    //Iterate through each object and write out its uri
+    if (defined(objects)) {
+        var ids = Object.keys(objects);
+        async.each(ids, function(id, asyncCallback) {
+            var object = objects[id];
+            if (defined(object.extras) && defined(object.extras,source)) {
+                var source = object.extras.source;
+                delete object.extras.source;
+
+                //Write the source object as a data or file uri depending on the isEmbedded flag
+                if (isEmbedded) {
+                    var sourceDataUri = new dataUri();
+                    switch (name) {
+                        case 'images':
+                            sourceDataUri.format('.png', source);

```

Comment:
Valid glTF images are `png`, `jpg`, `gif`, and `bmp` (since these are the ones supported by browsers).  Perhaps we should save off the format when loading so that we can write out the same image format when writing.

Later, we'll have options to convert images (e.g., convert all images to `jpg`), resize images, etc.